src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -162,103 +169,105 @@ public void validateIncompatibleJavadocCheckNames() throws IOException {
-                        .that(SIMPLE_CHECK_NAMES)
-                        .contains(check);
-
-                assertWithMessage(
-                            "Check '" + check + "' is now compatible with SuppressionXpathFilter."
-                                + " Please update the todo list in"
-                                + " XpathRegressionTest.INCOMPATIBLE_CHECK_NAMES")
-                        .that(INCOMPATIBLE_CHECK_NAMES.contains(check))
-                        .isFalse();
-                compatibleChecks.add(check);
-            }
-        }
-
         // Ensure that all lists are up-to-date
         final Set<String> allChecks = new HashSet<>(SIMPLE_CHECK_NAMES);
         allChecks.removeAll(INCOMPATIBLE_JAVADOC_CHECK_NAMES);
         allChecks.removeAll(INCOMPATIBLE_CHECK_NAMES);
         allChecks.removeAll(Set.of("Regexp", "RegexpSinglelineJava", "NoCodeInFile"));
         allChecks.removeAll(NO_VIOLATION_MODULES);
-        allChecks.removeAll(compatibleChecks);
+        allChecks.removeAll(compatibleChecks());
         allChecks.removeAll(INTERNAL_MODULES);
 
         assertWithMessage("XpathRegressionTest is missing for [" + String.join(", ", allChecks)
                 + "]. Please add them to src/it/java/org/checkstyle/suppressionxpathfilter")
-                        .that(allChecks)
-                        .isEmpty();
+                .that(allChecks)
+                .isEmpty();
     }
 
-    @Test
-    public void validateInputFiles() throws Exception {
-        try (DirectoryStream<Path> dirs = Files.newDirectoryStream(inputDir)) {
-            for (Path dir : dirs) {
-                // input directory must be named in lower case
-                assertWithMessage(dir + " is not a directory")
-                        .that(Files.isDirectory(dir))
-                        .isTrue();
-                final String dirName = dir.toFile().getName();
-                assertWithMessage("Invalid directory name: " + dirName)
-                        .that(ALLOWED_DIRECTORY_AND_CHECKS)
-                        .containsKey(dirName);
+    private Set<String> compatibleChecks() throws IOException {
+        try (Stream<Path> files = Files.list(javaDir)) {
+            return files
+                    .filter(Files::isRegularFile)
+                    .map(Path::getFileName)
+                    .map(Path::toString)
+                    .filter(name -> !name.startsWith("Abstract"))
+                    .flatMap(name -> firstGroupMatchOrEmpty(NAME.matcher(name)))
+                    .peek(test -> {
+                        if (!SIMPLE_CHECK_NAMES.contains(test)) {
+                            throw new IllegalStateException("Unknown test '"
+                                    + test + "' in test file: " + test);
+                        }
+                        if (INCOMPATIBLE_CHECK_NAMES.contains(test)) {
+                            throw new IllegalStateException("Test '"
```

Comment:
```suggestion
                            throw new IllegalStateException("Check '"
```
i do not see how it is enforce but will undo.